src/noInvalidFeature/noInvalidFeature.js

```diff
@@ -0,0 +1,59 @@
+import { load } from "js-yaml";
+import { readFileSync } from "fs";
+
+const meta = {
+  type: "problem",
+  docs: {
+    description: "Disallows features in flipper.d.ts if they don't exist in features.yml",
+    category: "no-invalid-feature",
+    recommended: false,
+  },
+  fixable: null,
+  schema: [
+    {
+      type: "object",
+      properties: {
+        staticFeaturesFilePath: {
+          type: "string",
+        },
+      },
+      required: ["staticFeaturesFilePath"],
+      additionalProperties: false,
+    },
+  ],
+};
+
+let validFeatures;
+const readFeatureYml = staticFeaturesFilePath => {
+  if (validFeatures) return;
+  validFeatures = load(readFileSync(staticFeaturesFilePath, "utf8"));
+};
+
+/**
+ * Looks through flipper.d.ts and ensures that all flippers listed appear in features.yml
+ *
+ * @param {import("@types/eslint").Rule.RuleContext} context
+ * @returns {import("@typescript-eslint/utils/dist/index").ESLintUtils.RuleListener}
+ */
+const create = context => {
+  if (!context.getFilename().endsWith("flipper.d.ts")) return {};
```

Comment:
If we aren't processing the `flipper.d.ts` file, let's not do anything. Otherwise, read the contents of the `features.yml` file (passed via configuration) and compare contents of flipper.d.ts to it.

If a feature in `flipper.d.ts` does not exist in `features.yml`, error out.